nixos/modules/programs/foot/default.nix

```
@@ -0,0 +1,79 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.programs.foot;
+
+  settingsFormat = pkgs.formats.iniWithGlobalSection { listsAsDuplicateKeys = true; };
+in
+{
+  options.programs.foot = {
+    enable = lib.mkEnableOption "foot terminal emulator";
+
+    package = lib.mkPackageOption pkgs "foot" { };
+
+    settings = lib.mkOption {
+      inherit (settingsFormat) type;
+      default = {
+        globalSection = { };
+      };
+      description = ''
+        Configuration for foot terminal emulator. Further information can be found in foot's man page foot.ini(5).
+      '';
+    };
+
+    theme = lib.mkOption {
+      type = with lib.types; nullOr str;
+      default = null;
+      description = ''
+        Include theme from https://codeberg.org/dnkl/foot/src/branch/master/themes.
+      '';
+    };
+
+    enableBashIntergration = lib.mkEnableOption "foot bash integration";
+    enableFishIntergration = lib.mkEnableOption "foot fish integration";
+    enableZshIntergration = lib.mkEnableOption "foot zsh integration";
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment = {
+      systemPackages = [ cfg.package ];
+      etc."xdg/foot/foot.ini".text = lib.generators.toINIWithGlobalSection {
+        listsAsDuplicateKeys = true;
+        mkKeyValue =
+          with lib.generators;
+          mkKeyValueDefault {
+            mkValueString =
+              v:
+              mkValueStringDefault { } (
+                if v == true then
+                  "yes"
+                else if v == false then
+                  "no"
+                else if v == null then
+                  "none"
+                else
+                  v
+              );
+          } "=";
+      } cfg.settings;
+    };
+    programs = {
+      foot.settings.globalSection.include = lib.optionals (cfg.theme != null) [
+        "${pkgs.foot.themes}/share/foot/themes/${cfg.theme}"
+      ];
+      # https://codeberg.org/dnkl/foot/wiki#user-content-shell-integration
+      bash.interactiveShellInit = lib.mkIf cfg.enableBashIntergration ". ${./bashrc}";
+      fish.interactiveShellInit = lib.mkIf cfg.enableFishIntergration ". ${./config.fish}";
```

```suggestion
      fish.interactiveShellInit = lib.mkIf cfg.enableFishIntergration "source ${./config.fish}";
```


> . (a single period) is an alias for the source command. The use of . is deprecated in favour of source, and . will be removed in a future version of fish.

https://fishshell.com/docs/current/cmds/source.html